openmdao/core/tests/test_system.py

```diff
@@ -255,6 +256,76 @@ def test_deprecated_metadata(self):
         prob.run_model()
         assert_rel_error(self, prob['double.y'], [2., 4., 6.])
 
+    def test_list_inputs_output_with_includes_excludes(self):
+        from openmdao.test_suite.scripts.circuit_analysis import Resistor, Diode, Node, Circuit
+
+        p = Problem()
+        model = p.model
+
+        model.add_subsystem('ground', IndepVarComp('V', 0., units='V'))
+        model.add_subsystem('source', IndepVarComp('I', 0.1, units='A'))
+        model.add_subsystem('circuit', Circuit())
+
+        model.connect('source.I', 'circuit.I_in')
+        model.connect('ground.V', 'circuit.Vg')
+
+        p.setup()
+        p.run_model()
+
+        # Inputs with no includes or excludes
+        stream = cStringIO()
+        model.list_inputs(out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 11)  # First word is the # of inputs
+
+        # Inputs with includes
+        stream = cStringIO()
+        model.list_inputs(includes=['*V_out*'], out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 3)
+
+        # Inputs with includes matching a promoted name
+        stream = cStringIO()
+        model.list_inputs(includes=['*Vg*'], out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 2)
+
+        # Inputs with excludes
+        stream = cStringIO()
+        model.list_inputs(excludes=['*V_out*'], out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 8)
+
+        # Inputs with excludes matching a promoted name
+        stream = cStringIO()
+        model.list_inputs(excludes=['*Vg*'], out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 9)
+
+        # Inputs with includes and excludes
+        stream = cStringIO()
+        model.list_inputs(includes=['*V_out*'], excludes=['*Vg*'], out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 1)
+
+
+        # Outputs with no includes or excludes. Explicit only
+        stream = cStringIO()
+        model.list_outputs(implicit=False, out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 5)  # First word is the # of outputs
+
+        # Outputs with includes. Explicit only
+        stream = cStringIO()
+        model.list_outputs(includes=['*I'], implicit=False, out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 4)
+
+        # Outputs with excludes. Explicit only
+        stream = cStringIO()
+        model.list_outputs(excludes=['circuit*'], implicit=False, out_stream=stream)
+        text = stream.getvalue()
+        self.assertEqual(int(text.split()[0]), 2)
```

Comment:
it would probably be more efficient to set out_stream to None and just check the length of the return value

Reply:
I like that ! Will do